secret/secrets_manager_vault_test.go

```diff
@@ -0,0 +1,133 @@
+package secret
+
+import (
+	"context"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/evergreen-ci/cocoa/awsutil"
+	"github.com/evergreen-ci/utility"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSecretsManager(t *testing.T) {
+	assert.Implements(t, (*Vault)(nil), &BasicSecretsManager{})
+
+}
+
+func TestVaultCreateAndDeleteSecret(t *testing.T) {
+	checkAWSEnvVars(t)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	hc := utility.GetHTTPClient()
+	defer utility.PutHTTPClient(hc)
+
+	c, err := NewBasicSecretsManagerClient(awsutil.ClientOptions{
+		Creds:  credentials.NewEnvCredentials(),
+		Region: aws.String(os.Getenv("AWS_REGION")),
+		Role:   aws.String(os.Getenv("AWS_ROLE")),
+		RetryOpts: &utility.RetryOptions{
+			MaxAttempts: 5,
+		},
+		HTTPClient: hc,
+	})
+	require.NoError(t, err)
+
+	m := NewBasicSecretsManager(c)
+
+	Cleanup := func(out string) {
+		if out != "" {
+			err := m.DeleteSecret(ctx, out)
+			require.NoError(t, err)
+		}
+	}
+
+	// Temporarily commented out because waiting for input validation
+	// t.Run("VaultCreateFailsWithInvalidInput", func(t *testing.T) {
+	// 	out, err := m.CreateSecret(ctx, NamedSecret{})
+	// 	assert.Error(t, err)
+	// 	assert.Zero(t, out)
+	// })
+
+	t.Run("VaultDeleteFailsWithInvalidInput", func(t *testing.T) {
```

Comment:
"DeleteFailsWithInvalidInput"

You don't need to include the word "Vault" in the test name because it's already implied that you're testing the vault in the function's name.